### PR TITLE
feat(config): add discoverable PawWork home

### LINKED ISSUE #328: [Feature] Add discoverable PawWork home for global user config
## What task are you trying to do?

Users need a predictable, easy-to-find place for PawWork-wide personal configuration.

The concrete task is: create or edit global instructions/configuration that should apply across all PawWork sessions, especially files such as:

- global `AGENTS.md`
- global `pawwork.json`
- user-level skills / commands
- user-level MCP-related configuration

These are human-editable files, not runtime-managed app data.

## What do you do today?

Today PawWork's global `AGENTS.md` is loaded from the app runtime config tree. On macOS, that currently resolves to a path like:

```text
~/Library/Application Support/ai.pawwork.desktop/config/pawwork/AGENTS.md
```

That follows Electron/platform app-data conventions, but it is painful for this workflow because users have to either:

- know a deep platform-specific runtime path,
- ask PawWork where the global `AGENTS.md` is loaded from,
- set `PAWWORK_CONFIG_DIR` manually,
- or keep project-level `AGENTS.md` files even when the rule is really global.

It also mixes two different categories of files:

- human-edited, syncable config;
- runtime-managed app data/cache/state/logs.

That makes the config harder to discover, harder to document, and awkward to manage with dotfiles.

## What would a good result look like?

PawWork should have a clear global user configuration home for files that humans are expected to edit and possibly sync across machines.

A good result would let users reason about global config with a simple model, for example:

```text
~/.pawwork/
  AGENTS.md
  pawwork.json
  skills/
  commands/
  mcp/
```

Runtime-managed files should remain in platform app data/cache/state locations, for example:

- session databases
- logs
- caches
- tool output
- snapshots / undo state
- Chromium/session data

Possible acceptance criteria:

- Add an explicit global config home concept, e.g. `PAWWORK_HOME`.
- Default the human-editable config home to a discoverable path such as `~/.pawwork`.
- Continue supporting the existing platform config path for backward compatibility.
- Define clear precedence for global `AGENTS.md` and `pawwork.json`.
- Avoid moving cache/log/database/session data into the human-editable config home.
- Provide an app command or UI entry like “Open global config folder”.
- Document the distinction between global user config, project config, runtime data, and enterprise/managed config.

One possible precedence model for global instructions:

```text
1. PAWWORK_HOME/AGENTS.md
2. ~/.pawwork/AGENTS.md
3. existing platform config AGENTS.md
4. project AGENTS.md / CLAUDE.md / CONTEXT.md
```

## Which audience does this matter to most?

Both

## Extra context

PawWork is cross-platform, not macOS-only. The current platform app-data layout is useful for runtime data, but less discoverable for developer/agent-style global config.

Relevant prior art:

- Claude Code uses user scope at `~/.claude/`, including `~/.claude/CLAUDE.md`, `~/.claude/settings.json`, and `~/.claude/agents/`.
- Codex uses `~/.codex/config.toml` for user-level config and `.codex/config.toml` for project overrides.
- GitHub CLI exposes `GH_CONFIG_DIR` and defaults to XDG/platform config locations.
- VS Code stores user settings in platform locations, but exposes commands to open the settings JSON directly.
- Electron's `userData` path is suitable for app configuration/runtime data, but Electron also warns not to put large files there because some environments back it up.

The key product distinction should be:

- **Human-editable global config**: easy to find, document, and sync.
- **Runtime-managed app data**: platform-appropriate app data/cache/state locations.
- **Secrets/tokens**: credential store/keychain where possible; not in `AGENTS.md`.

### PR DESCRIPTION
**Pull request by Astro-Han**
## Summary

Add a PawWork Home for human-editable global config and wire global config, instructions, resources, CLI writes, plugin writes, path reporting, and session export sources through it.

Follow-up fixes address the review feedback: `unit-opencode` regression coverage, explicit Home creation, cross-platform env path handling, legacy config seeding before global MCP/plugin writes, aligned global config write resolution, shared PawWork Home resolver helpers, MCP project config resolution, session export provenance, command-palette error reporting, CLI seeding error UX, PAWWORK_HOME relative instruction resolution, directory-only resource discovery, formal spec docs, dynamic plugin global path resolution, and OpenCode `/path?ensureConfig=true` behavior.

## Why

Fixes #328. PawWork needs a discoverable global config folder for user-edited files such as `AGENTS.md`, `pawwork.json(c)`, commands, agents, skills, and plugin/MCP config, while runtime data stays in platform app data/cache/state/log locations.

## Related Issue

Fixes #328

## Human Review Status

Pending. A human should make the final merge decision after reviewing the final diff and verification evidence.

## Review Focus

Please focus on migration behavior: `PAWWORK_HOME` precedence, legacy `Global.Path.config` fallback, first-write preservation of existing legacy config, OpenCode-mode isolation, explicit versus read-only config path handling, session export `instruction_sources`, global MCP/plugin write consistency, local MCP project config write targets, and runtime-env-sensitive path resolution.

## Risk Notes

This changes PawWork global config lookup and write targets. Runtime data paths are intentionally unchanged. Legacy platform config remains a read fallback, and first writes to the new Home preserve the effective legacy config to avoid making old settings appear lost.

## Docs

The PawWork Home behavior is documented in `packages/opencode/specs/pawwork-home.md`. It is intentionally not in the root README.

## How To Verify

```text
Focused seed/provenance regression tests: 69 passed, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/session/export.test.ts --timeout 30000

Focused plugin/instruction regression tests: 72 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/plugin/install.test.ts test/plugin/install-concurrency.test.ts test/session/instruction.test.ts --timeout 30000

OpenCode package typecheck after seed/provenance follow-up: passed
Command: bun --cwd packages/opencode typecheck

Core package typecheck after PawWorkHome cleanup: passed
Command: bun --cwd packages/core typecheck

Diff check after seed/provenance follow-up: passed
Command: git diff --check

Focused config boundary review tests: 138 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/plugin/install.test.ts test/session/export.test.ts test/session/instruction.test.ts --timeout 30000

Plugin install concurrency regression tests: 3 passed, 0 failed
Command: bun --cwd packages/opencode test test/plugin/install-concurrency.test.ts --timeout 30000

OpenCode package typecheck after config boundary follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after config boundary follow-up: passed
Command: git diff --check

Focused latest CodeRabbit edge-case regression tests: 238 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/plugin/install.test.ts test/plugin/install-concurrency.test.ts test/session/instruction.test.ts test/session/export.test.ts test/config/config.test.ts test/config/pawwork-global-config.test.ts --timeout 30000

OpenCode package typecheck after latest CodeRabbit edge-case follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after latest CodeRabbit edge-case follow-up: passed
Command: git diff --check

Focused config write serialization tests: 174 passed, 0 failed
Command: bun --cwd packages/opencode test test/plugin/install.test.ts test/plugin/install-concurrency.test.ts test/config/config.test.ts test/config/pawwork-global-config.test.ts --timeout 30000

OpenCode package typecheck after config write serialization follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after config write serialization follow-up: passed
Command: git diff --check

Focused legacy write-boundary regression tests: 96 passed, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/plugin/install.test.ts test/session/export.test.ts --timeout 30000

OpenCode package typecheck after legacy write-boundary follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after legacy write-boundary follow-up: passed
Command: git diff --check

Focused fallback precedence regression tests: 72 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/session/instruction.test.ts --timeout 30000

Focused PawWork global config tests: 37 passed, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts --timeout 30000

Focused instruction tests: 35 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/session/instruction.test.ts --timeout 30000

OpenCode package typecheck after fallback precedence follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after fallback precedence follow-up: passed
Command: git diff --check

Focused instruction provenance tests: 60 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/session/instruction.test.ts test/session/export.test.ts --timeout 30000

OpenCode package typecheck after instruction provenance follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after instruction provenance follow-up: passed
Command: git diff --check


Focused plugin/path review follow-up tests: 34 passed, 0 failed
Command: bun --cwd packages/opencode test test/plugin/install.test.ts test/server/default-directory.test.ts --timeout 30000

OpenCode package typecheck after plugin/path review follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after plugin/path review follow-up: passed
Command: git diff --check


Focused CodeRabbit follow-up tests: 65 passed, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/server/default-directory.test.ts test/plugin/install.test.ts --timeout 30000

OpenCode package typecheck after CodeRabbit follow-up: passed
Command: bun --cwd packages/opencode typecheck

Diff check after CodeRabbit follow-up: passed
Command: git diff --check


Focused runtime resolution regression tests: 64 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/session/instruction.test.ts test/plugin/install.test.ts test/server/default-directory.test.ts

OpenCode package typecheck after runtime resolution fixes: passed
Command: bun --cwd packages/opencode typecheck

Focused resource-directory regression tests: 34 passed, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts

OpenCode package typecheck after resource-directory fix: passed
Command: bun --cwd packages/opencode typecheck

Focused latest review plus CodeRabbit regression tests: 115 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/session/export.test.ts test/plugin/install.test.ts test/session/instruction.test.ts

OpenCode package typecheck after CodeRabbit follow-up: passed
Command: bun --cwd packages/opencode typecheck

Focused latest review regression tests: 83 passed, 0 failed
Command: bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/session/export.test.ts test/plugin/install.test.ts

OpenCode package typecheck after latest review fixes: passed
Command: bun --cwd packages/opencode typecheck

Focused PawWork Home, plugin write, and export provenance tests: 80 passed, 0 failed
Command: bun --cwd packages/opencode test test/plugin/install.test.ts test/session/export.test.ts test/config/pawwork-global-config.test.ts

Focused PawWork Home, instruction, plugin write, and export provenance tests: 109 passed, 1 existing todo, 0 failed
Command: bun --cwd packages/opencode test test/session/export.test.ts test/session/instruction.test.ts test/config/pawwork-global-config.test.ts test/plugin/install.test.ts

OpenCode package typecheck: passed
Command: bun --cwd packages/opencode typecheck

App package typecheck: passed
Command: bun --cwd packages/app typecheck

Previous full opencode unit suite: 2515 passed, 9 skipped, 1 todo, 0 failed
Command: bun --cwd packages/opencode test

Previous SDK typecheck: passed
Command: bun --cwd packages/sdk/js typecheck

Previous core typecheck: passed
Command: bun --cwd packages/core typecheck

Diff check: passed
Command: git diff --check
```

## Screenshots or Recordings

Not captured. The UI change is a command palette entry that explicitly asks `/path?ensureConfig=true` before opening the global config folder and now shows an error toast if Home creation or system open fails; behavior is covered by server path regression tests and app typecheck.

## Checklist

- [x] Human review status is stated above as pending, approved, or not required
- [x] I linked the related issue, or stated why there is no issue
- [x] This PR has type, scope, and priority labels, or I requested maintainer labeling
- [x] I described the review focus and any meaningful risks
- [x] I listed the relevant verification steps and the key result for each
- [x] I did not introduce unrelated refactors, dependencies, generated files, or file changes beyond the stated scope
- [x] I manually checked visible UI or copy changes when needed, with screenshots or recordings
- [x] I considered macOS and Windows impact for desktop, packaging, updater, signing, paths, shell, or permissions changes
- [x] I called out docs, release notes, dependencies, permissions, credentials, deletion behavior, generated content, or local file changes when relevant
- [x] I reviewed the final diff for unrelated changes and suspicious dependency changes
- [x] I am targeting `dev`, and my PR title and commit messages use Conventional Commits in English

## Summary by CodeRabbit

* **New Features**
  * "Open global config folder" command in settings; runtime-aware global config resolution, ensure-config option, atomic global config writes, and seeding behavior for first-write migration.

* **Localization**
  * Added English and Chinese translations for the command and its failure toast.

* **Documentation**
  * Added spec describing PawWork Home behavior and resolution priority.

* **Tests**
  * Expanded coverage for PawWork Home, config resolution, plugin install flows, server path handling, and instruction-source provenance.

### CODE REVIEW

**Comment by Astro-Han** [comment]
@coderabbitai full review

File: packages/opencode/src/config/config.ts [hunk 1]
```diff
@@ -364,13 +366,40 @@ export interface Interface {
 export class Service extends Context.Service<Service, Interface>()("@opencode/Config") {}
 
 function globalConfigFile() {
+  if (Runtime.isPawWork()) {
+    const primary = PawWorkHome.primary()
+    const jsonc = path.join(primary, "pawwork.jsonc")
+    if (existsSync(jsonc)) return jsonc
+    const json = path.join(primary, "pawwork.json")
+    if (existsSync(json)) return json
+    return json
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/config/config.ts]: Fixed in 6218721ad. I exposed `Config.globalConfigFileForWrite()` and switched global MCP writes to that resolver, so PawWork global writes no longer scan or select `opencode.json(c)`. The resolver continues to use `PawWorkHome`, which is based on `Global.Path.home` for deterministic tests.

File: packages/opencode/src/config/config.ts [hunk 2]
```diff
@@ -857,7 +882,11 @@ const rawLayer = Layer.effect(
 
     const updateGlobal = Effect.fn("Config.updateGlobal")(function* (config: Info) {
       const file = globalConfigFile()
-      const before = (yield* readConfigFile(file)) ?? "{}"
+      yield* Effect.promise(() =>
+        Runtime.isPawWork() ? PawWorkHome.ensurePrimary() : fsNode.mkdir(path.dirname(file), { recursive: true }),
+      )
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/config/config.ts]: Addressed the semantic part in 6218721ad by adding explicit `Config.seedGlobalConfig()` and using it from MCP/plugin global writes instead of `Config.updateGlobal({})`. I did not add parameter injection for `Runtime.isPawWork()` because this branch is product namespace selection, not OS platform-dependent path handling; the cross-platform path behavior is covered by the PawWork Home tests.

File: packages/opencode/src/cli/cmd/mcp.ts [hunk 3]
```diff
@@ -408,31 +410,30 @@ export const McpLogoutCommand = cmd({
...
   for (const candidate of candidates) {
     if (await Filesystem.exists(candidate)) {
       return candidate
     }
   }
 
-  // Default to pawwork.json if none exist.
-  return candidates[0]
+  return Runtime.isPawWork() ? path.join(baseDir, "pawwork.json") : path.join(baseDir, "opencode.json")
 }
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/cli/cmd/mcp.ts]: Fixed in cb2139175 by switching MCP local config resolution to `Config.projectConfigFileForWrite()`. In OpenCode mode it now follows the same project write resolver as `Config.update()` and ignores `pawwork.json(c)` aliases; regression added for a project containing both `pawwork.json` and `opencode.json`.

File: packages/opencode/src/cli/cmd/mcp.ts [hunk 4]
```diff
@@ -528,6 +530,7 @@ export const McpAddCommand = cmd({
             command: command.split(" "),
           }
 
+          if (Runtime.isPawWork() && configPath === globalConfigPath) await Config.seedGlobalConfig()
           await addMcpToConfig(name, mcpConfig, configPath)
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/cli/cmd/mcp.ts]: Fixed in cb2139175. PawWork global MCP seeding now runs through `seedGlobalConfigIfNeeded()`, catches initialization failures, prints a CLI error, exits the add flow, and only writes JSONC when seeding succeeds.

File: packages/opencode/src/cli/cmd/plug.ts [hunk 5]
```diff
@@ -129,6 +134,7 @@ export function createPlugTask(input: PlugInput, dep: PlugDeps = defaultPlugDeps
 
     const patch = dep.spinner()
     patch.start("Updating plugin config...")
+    if (global && Runtime.isPawWork()) await dep.seedGlobalConfig?.()
     const out = await patchPluginConfig(
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/cli/cmd/plug.ts:154]: Fixed in cb2139175. `createPlugTask()` now catches `dep.seedGlobalConfig?.()` failures before `patchPluginConfig`, stops the patch spinner with a failure state, logs the error through the existing CLI error path, returns `false`, and leaves the config untouched. Covered by `returns false when PawWork global plugin config seeding fails`.

File: packages/opencode/src/server/instance/middleware.ts [hunk 6]
```diff
@@ -69,6 +71,9 @@ export function WorkspaceRouterMiddleware(upgrade: UpgradeWebSocket): Middleware
     const directory = Filesystem.resolve(decoded)
 
     const url = new URL(c.req.url)
+    if (url.pathname === "/path" && url.searchParams.get("ensureConfig") === "true" && !Runtime.isPawWork()) {
+      mkdirSync(Global.Path.config, { recursive: true })
+    }
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/server/instance/middleware.ts]: Handled in edc4e5fae. I verified the preferred delete-only option and it breaks the OpenCode `/path?ensureConfig=true` regression because `Instance.provide` runs before the route handler. Kept the early creation but wrapped it in try/catch so ENOTDIR or permission failures do not escape the middleware; the handler still owns the final config creation/error path. Covered by the OpenCode path route regression.

File: packages/opencode/test/server/default-directory.test.ts [hunk 7]
```diff
@@ -34,4 +35,88 @@ describe("default directory routing", () => {
...
+    } finally {
+      if (previousRuntime === undefined) delete process.env.PAWWORK_RUNTIME_NAMESPACE
+      else process.env.PAWWORK_RUNTIME_NAMESPACE = previousRuntime
+      if (previousHome === undefined) delete process.env.OPENCODE_TEST_HOME
+      else process.env.OPENCODE_TEST_HOME = previousHome
+      if (previousPawWorkHome === undefined) delete process.env.PAWWORK_HOME
+      else process.env.PAWWORK_HOME = previousPawWorkHome
+      ;(Global.Path as { config: string }).config = previousConfig
+    }
+  })
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/test/server/default-directory.test.ts:103]: Fixed in edc4e5fae. The two PawWork `/path` tests now mock `os.homedir()` with the temp directory and restore it in `finally`, so the middleware and instance bootstrap cannot touch the real `~/PawWork` during the tests.

File: packages/opencode/src/plugin/install.ts [hunk 8]
```diff
@@ -320,58 +323,86 @@ export async function readPluginManifest(target: string): Promise<ManifestResult
 }
 
 function patchDir(input: PatchInput) {
-  if (input.global) return input.config ?? Global.Path.config
+  if (input.global) return input.config ?? (Runtime.isPawWork() ? PawWorkHome.primary() : Global.Path.config)
   const git = input.vcs === "git" && input.worktree !== "/"
   const root = git ? input.worktree : input.directory
+  if (Runtime.isPawWork()) return path.dirname(Config.projectConfigFileForWrite(root))
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/plugin/install.ts:329]: Fixed in `935302425`: `patchPluginConfig()` now seeds PawWork global config before a direct global patch when no explicit config path is supplied, matching the CLI/MCP first-write behavior so legacy effective config is copied into Home before the plugin patch is applied. Added regression coverage for direct global patching from legacy fallback.

File: packages/opencode/src/session/export.ts [hunk 9]
```diff
@@ -298,23 +298,23 @@ export namespace Export {
...
+    const loaded = (yield* instruction.sources({ fetchRemote: false })).filter((source) => source.status === "loaded")
+
+    for (const source of loaded) {
+      if (source.kind === "remote") {
+        sources.push({ kind: source.kind, url: source.path, hash_unavailable: true })
+        continue
+      }
+
+      const hash = yield* Effect.promise(() => hashFile(source.path))
+      if (hash) sources.push({ kind: source.kind, path: source.path, hash })
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/session/export.ts]: Fixed in `935302425`: local instruction export provenance now keeps the source entry with `hash_unavailable: true` if hashing fails after `Instruction.sources()` reported it as loaded. This preserves evidence that the instruction contributed to prompt construction without weakening the hash contract.

File: packages/opencode/src/session/instruction.ts [hunk 10]
```diff
@@ -239,40 +247,44 @@ export const layer: Layer.Layer<Service, never, AppFileSystem.Service | Config.S
...
               path: resolved,
+              kind: "global",
               reason: "skipped because a higher-priority global instruction file was loaded",
             })
             continue
           }
-          const ok = yield* recordFileEntry(resolved)
+          const ok = yield* recordFileEntry(resolved, "global")
           if (ok) globalLoaded = true
+          break
```
[the remark this answers is not in the document]
Comment (Astro-Han) [reply packages/opencode/src/session/instruction.ts:307]: Fixed in `935302425`: global instruction discovery now requires `AGENTS.md` candidates to be regular files. Non-file candidates are reported as considered with reason `not a file` and lower-priority fallbacks continue to be evaluated. Added a regression for a directory named `AGENTS.md` in a higher-priority PawWork Home.

**Comment by Astro-Han** [comment]
Addressed the latest config seeding review in `b9b7442e6`.

What changed:
- First PawWork Home writes now seed from the raw legacy config source text instead of `JSON.stringify(loadGlobal())`, so `{file:...}` placeholders are preserved and secrets are not materialized into the new `pawwork.json`.
- Relative `instructions` and relative local plugin paths are rewritten against the legacy source file before being copied into Home, so migration does not silently rebase them under the new Home directory.
- Multiple legacy sources (`pawwork.json` + `pawwork.jsonc`) are merged from raw source text in load order before applying the update, preserving the prior effective config without using substituted runtime values.
- New Home config files inherit the active legacy source file mode; when no source exists, PawWork first write defaults to `0600`.
- Non-primary legacy / `PAWWORK_CONFIG_DIR` reads now strip deprecated `default_agent` in memory while leaving the legacy file unchanged.

Verification:
- `bun test ./test/config/pawwork-global-config.test.ts` from `packages/opencode` passes: 44 pass, 0 fail.
- `bun run typecheck` from `packages/opencode` passes.

Not folded into this commit:
- The resolver/lock consolidation, injectable `PawWorkHomeResolver`, and broader writer API cleanup are valid follow-up architecture work, but they are P2/P3 scope and would expand this PR beyond the current security/migration fix.

**Comment by Astro-Han** [comment]
Latest review follow-up:

Fixed in `a9800eb7d`:

- P1: legacy TOML migration now preserves existing `pawwork.json` / `opencode.json` by reading the target and merging TOML as the lower-priority source before deleting the TOML file. Added regression coverage for existing JSON + legacy TOML.
- P2: seed migration is now schema-aware. It only absolutizes top-level `instructions`, top-level plugin specs, and `{file:...}` placeholders. Nested business fields named `instructions` or `plugin` are no longer treated as config schema fields. Plugin tuple options still get placeholder-preserving `{file:...}` rewriting, and unresolved placeholders such as `{env:...}` in instructions are preserved.
- P2: session export now resolves `instruction_sources` from the exported root session directory instead of whichever `Instance.directory` is current at export time. Added a regression where current cwd differs from `session.directory`.
- P3: project instruction discovery now skips non-file `AGENTS.md` / `CLAUDE.md` / `CONTEXT.md` candidates instead of letting a directory block fallback.
- P3: PawWork Home candidate stat failures now only treat `ENOENT` / `ENOTDIR` as absent; other stat failures surface instead of silently falling back.
- P3: implicit `$schema` writeback now uses the same config-file lock plus atomic replace helper.

No review items were rejected in this round.

Verification:

```text
bun test test/config/pawwork-global-config.test.ts --timeout 30000
46 pass, 0 fail

bun test test/session/instruction.test.ts --timeout 30000
37 pass, 1 todo, 0 fail

bun test test/session/export.test.ts --timeout 30000
28 pass, 0 fail

bun --cwd packages/opencode typecheck
ok

bun --cwd packages/core typecheck
ok

bun test test/global.test.ts --timeout 30000 (packages/core)
3 pass, 0 fail

bun test test/global/runtime-namespace.test.ts --timeout 30000 (packages/opencode)
2 pass, 0 fail

bun --cwd packages/opencode test --timeout 30000
2556 pass, 9 skip, 1 todo, 0 fail

git diff --check
ok
```

**Comment by Astro-Han** [comment]
Handled the latest review in `6411376db`.

Fixed:
- `seedConfigValueFromSource()` now runs the same read-path normalization/schema validation before seeding, so deprecated legacy keys like `theme`, `keybinds`, and `tui` are not persisted into the new strict PawWork Home config.
- Legacy TOML migration now chooses the active legacy global config target and performs target existence/read/write decisions inside the config-file lock. If an active JSON/JSONC already exists, migration deletes the TOML without creating a lower-priority split JSON file.
- Session export no longer fails when the original session directory is gone; it falls back to best-effort global/bundled provenance and marks the missing project directory as `hash_unavailable`.
- PawWork Home candidate dedupe no longer lets an inaccessible low-priority candidate prevent a higher-priority Home from loading.
- Default plugin CLI writes now use the Config atomic writer path, preserving existing mode bits.
- Nearby instruction discovery now skips directories named `AGENTS.md`, and project source diagnostics records non-file candidates.

Regression coverage added:
- legacy deprecated keys first-write seed
- TOML + active JSONC migration target behavior
- missing session directory export
- default plugin write mode preservation
- nearby instruction directory fallback

Deferred / not implemented in this patch:
- The broad `ConfigMigration/ConfigWriteBoundary` architecture consolidation is a real maintainability direction, but it is larger than this review round and not needed to close the concrete data-loss/export/path issues above. I would keep that as a follow-up refactor after #438 lands, so this PR stops accumulating cross-cutting rewrites.

Verification:
- `bun test test/config/pawwork-global-config.test.ts test/session/instruction.test.ts test/session/export.test.ts test/plugin/install.test.ts --timeout 30000`
- `bun --cwd packages/opencode typecheck`
- `bun --cwd packages/core typecheck`
- `git diff --check`

**Comment by Astro-Han** [comment]
Handled the latest review in `40fc72dc7` and re-ranked the items as follows.

P0:
- None.

P1:
- Fixed: OpenCode legacy TOML migration no longer has the “works once, disappears next launch” behavior. TOML is now treated as lower priority than active JSON/JSONC for runtime merge, and in OpenCode mode non-overlapping TOML fields are persisted into the active regular config file under the same config-file lock. Added `opencode.jsonc + config` regression covering current load and next load.

P2:
- Fixed: legacy TOML migration target selection now only accepts regular files. Directory/non-file candidates are skipped, and a regression covers `opencode.jsonc` as a directory.
- Fixed: seed instruction rebasing now handles middle placeholders such as `rules/{env:NAME}.md`; placeholder-prefixed values like `{env:RULES}/a.md` remain unchanged. Added regression coverage.
- Partially fixed: session export fallback for unavailable session directories now preserves active global `config.instructions` local paths and remote URL provenance without reading current project config. I did not add full `kind + scope/sourceFile` provenance in this patch because that is a schema/architecture change, not required to close the concrete fallback loss.
- Fixed boundary: `PawWorkHome.existingResourceDirectories()` still skips inaccessible low-priority fallback dirs, but primary/explicit env Home stat errors now fail instead of silently loading legacy resources.

P3:
- Current CI status is now fully green for head `6411376db`; after this push CI will rerun on `40fc72dc7`.
- Fixed: legacy TOML migration no longer swallows all errors silently; failures now emit a config warning with the legacy path and error.
- Deferred: broader Instruction provenance model (`kind + scope/sourceFile`) remains valid architecture debt, but I am not folding that into #438 because the specific export fallback issue is covered above and the broader model would expand the PR again.

Verification:
- `bun test test/config/pawwork-global-config.test.ts test/session/export.test.ts test/session/instruction.test.ts test/plugin/install.test.ts --timeout 30000` -> 152 pass, 1 todo, 0 fail
- `bun --cwd packages/opencode typecheck`
- `bun --cwd packages/core typecheck`
- `git diff --check`

**Comment by Astro-Han** [comment]
Handled the latest review in `2de8684a6` and re-ranked the items.

P0:
- None.

P1:
- None.

P2:
- Fixed: export fallback global config instructions now pass through `ConfigVariable.substitute()` with the config file as source, so `{env:...}` and `{file:...}` are resolved the same way as normal config loading.
- Fixed: export fallback no longer marks local global `instructions` as loaded blindly. It expands glob patterns relative to the global config file, only records matched files, and only records files with readable content. This avoids literal `rules/*.md` / missing-file provenance.
- Fixed/clarified by regression: OpenCode TOML migration keeps TOML arrays low priority when an existing active config already has the same array field. This matches the effective `mergeDeep(legacy, existing)` semantics; it does not concat arrays in migration because normal global merge does not concat them here.
- Fixed: legacy TOML data is normalized through the config schema before being merged or persisted. Deprecated legacy keys are stripped, unrecognized top-level keys are dropped, and invalid known fields still warn/fail migration instead of writing a config that would fail strict load later.

P3:
- Fixed: `PawWorkHome.existingResourceDirectories()` now fail-fasts only for explicit `PAWWORK_HOME` / `PAWWORK_CONFIG_DIR`; default `~/.pawwork` resource stat errors can fall back to legacy resources.
- Fixed: TOML migration warnings include legacy path, target path, and action (`merge`, `create`, `skip-existing`, or `unlink`) for easier diagnosis.
- Deferred: moving `globalConfiguredInstructionSources()` fully into Instruction/Config remains the right architecture direction, but this patch keeps the change bounded to export fallback. The concrete drift issues are now covered by substitution + glob/file regression tests; a broader provenance API refactor should be a follow-up, not another expansion of #438.

Verification:
- `bun test test/config/pawwork-global-config.test.ts test/session/export.test.ts test/session/instruction.test.ts test/plugin/install.test.ts --timeout 30000` -> 152 pass, 1 todo, 0 fail
- `bun --cwd packages/opencode typecheck`
- `bun --cwd packages/core typecheck`
- `git diff --check`

**Comment by Astro-Han** [comment]
Follow-up on the latest review, pushed `6c98d26f5`.

P0: none.
P1: none.
P2 handled:
- Legacy TOML + existing legacy `pawwork.json(c)`: migration now patches TOML-only fields into the active JSON/JSONC target instead of skipping then unlinking. PawWork runtime also merges TOML into the current result when the active global config source is the legacy platform dir, so current load and next load stay consistent.
- Nested unknown config keys: `normalizeWritableConfig()` now prunes `unrecognized_keys` at the reported schema path, not just top-level keys.
- Migration warning path: TOML migration now caches target/action/unlinked state during the attempt and does not recompute target info in the catch path.
- Session export fallback: relative global `config.instructions` resolved only because the session directory is unavailable are exported as fallback/considered provenance with `hash_unavailable` and a reason, not as normal loaded runtime sources. Absolute, env/file-substituted absolute paths, and remote URLs remain represented.

P3 / deferred:
- I did not introduce a new formal `Instruction.sources({ scope: "global" ... })` API in this PR. That is broader architecture cleanup; the current change keeps the fallback bounded and covered by regression tests.

Verification:
- `bun run --cwd packages/opencode typecheck`
- `bun --cwd packages/opencode test test/config/pawwork-global-config.test.ts test/session/export.test.ts`
- `bun run --cwd packages/opencode test`
